extensions/amp-tiktok/0.1/amp-tiktok.js

```diff
@@ -0,0 +1,253 @@
+              fill="#00F2EA"
+            />
+          </g>
+          <defs>
+            <clipPath id="clip0">
+              <rect width="44" height="44" fill="white" />
+            </clipPath>
+          </defs>
+        </svg>
+      `,
+    };
+  }
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!Element} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.videoId_ = null;
+
+    /** @private {?Function}*/
+    this.unlistenMessage_ = null;
+
+    /** @private {?Promise} */
+    this.resolveRecievedFirstMessage_ = null;
+
+    /** @private {string} */
+    this.iframeNameString_ = this.getIframeNameString_();
+
+    this.resizeOuter_ = debounce(
+      this.win,
+      (height) => {
+        resetStyles(this.iframe_, [
+          'width',
+          'height',
+          'position',
+          'opacity',
+          'pointer-events',
+        ]);
+        this.iframe_.removeAttribute('aria-hidden');
+        this.iframe_.setAttribute('aria-label', 'Tiktok');
+        this.iframe_.classList.remove('i-amphtml-tiktok-unresolved');
+        this.iframe_.classList.add('i-amphtml-tiktok-centered');
+        this.forceChangeHeight(height);
+      },
+      1000
+    );
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    //See
+    //https://developers.tiktok.com/doc/Embed
+    Services.preconnectFor(this.win).url(
+      this.getAmpDoc(),
+      'https://www.tiktok.com',
+      opt_onLayout
+    );
+  }
+
+  /** @override */
+  buildCallback() {
+    const {src} = this.element.dataset;
+    const videoIdRegex = /^((.+\/)?)(\d+)\/?$/;
+    if (src) {
+      this.videoId_ = src.replace(videoIdRegex, (_, _1, _2, id) => id);
+    } else {
+      const blockquoteOrNull = childElementByTag(this.element, 'blockquote');
+      if (
+        !blockquoteOrNull ||
+        !blockquoteOrNull.hasAttribute('placeholder') ||
+        !blockquoteOrNull.dataset.videoId
+      ) {
+        return;
+      }
+      this.videoId_ = blockquoteOrNull.dataset.videoId;
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    const {locale = 'en-US'} = this.element.dataset;
+    const src = `https://www.tiktok.com/embed/v2/${encodeURIComponent(
+      this.videoId_
+    )}?lang=${encodeURIComponent(locale)}`;
+
+    const iframe = createElementWithAttributes(
+      this.element.ownerDocument,
+      'iframe',
+      {
+        'src': src,
+        'name': this.iframeNameString_,
+        'aria-hidden': 'true',
+        'frameborder': '0',
+      }
+    );
+
+    iframe.classList.add('i-amphtml-tiktok-unresolved');
```

Comment:
Is it possible to make the default classname attribute during creation?

`{ ..., class|className: 'i-amphtml-...'}`